### ISSUE TITLE
`image_viewer`: beam center position is shifted by (0.5, 0.5)

### ISSUE THREAD

**Issue by elena-pascal** [dials/dials#2190]
The beam center in image_viewer is shown with a different definition of pixel positions, ie shifted by (-0.5, -0.5) wrt to the rest of the image

This is a direct beam image
![beam_center](https://user-images.githubusercontent.com/30300213/182590003-df745586-0d61-4a83-a042-8c8344bbb9ad.png)

The beam position saved is actually correct
![correct_beam_center](https://user-images.githubusercontent.com/30300213/182591130-9461f557-bd7c-427c-84d3-769e47d04760.png)

Here is what happens if I set image origin to (0, 0) as opposed to (-0.5, -0.5)
![wrong_beam_center](https://user-images.githubusercontent.com/30300213/182591415-edd3f846-75fe-463f-9725-57c3039d240f.png)

**Comment by dagewa** [comment]
See also https://github.com/dials/dials/issues/1545

**Comment by elena-pascal** [comment]
What is the intended behaviour? Center of pixel is integer or edge of pixel is integer?

**Comment by dagewa** [comment]
I think edge is integer is correct, so top left corner should be 0,0

**Comment by elena-pascal** [comment]
bottom left?

**Comment by elena-pascal** [comment]
If pixel edges are desired to be integers, that is not what is happening at the moment.

**Comment by dagewa** [comment]
> bottom left?

I meant top left in the `dials.image_viewer` (for most single panel detectors), which should read out `fast=0 / slow=0` but actually has both at `-0.5`

**Comment by phyy-nx** [comment]
I think the readout of -0.5 for the corner of the 0,0 pixel is correct, sinc in dials the center of the pixel is defined as the 0,0 position in pixel and millimeter space, right?

**Comment by nksauter** [comment]
A separate but related issue is that the mouse-over pixel intensity readout
is half pixel out of sync.  Would appreciate this diff being added to the
code:

*diff --git a/src/dials/util/image_viewer/slip_viewer/frame.py
b/src/dials/util/image_viewer/slip_viewer/frame.py*

*index 736c4e007..c10c7c92d 100644*

*--- a/src/dials/util/image_viewer/slip_viewer/frame.py*

*+++ b/src/dials/util/image_viewer/slip_viewer/frame.py*

@@ -222,8 +222,8 @@ def handle_position_event(self, event):

                 possible_intensity = None

                 fi = self.pyslip.tiles.raw_image

                 detector = fi.get_detector()

-                ifs = (int(coords[1]), int(coords[0]))  # int fast slow

-                isf = (int(coords[0]), int(coords[1]))  # int slow fast

+                ifs = (int(coords[1]+0.5), int(coords[0]+0.5))  # int fast
slow

+                isf = (int(coords[0]+0.5), int(coords[1]+0.5))  # int slow
fast

                 image_data = fi.get_image_data()

                 if not isinstance(image_data, tuple):

                     image_data = (image_data,)

**Comment by phyy-nx** [comment]
@nksauter great minds, etc.  I'm prepping this exact PR right now :)

**Comment by dagewa** [comment]
> I think the readout of -0.5 for the corner of the 0,0 pixel is correct, since in dials the center of the pixel is defined as the 0,0 position in pixel and millimeter space, right?

I don't think this is correct. Please see my comments on the PR: https://github.com/dials/dials/pull/2194#issuecomment-1207918034

Very glad we are getting somewhere with sorting the bug(s) out though
